### PR TITLE
tests,tools: Update t8n to lightclient/devnet-8 version

### PR DESCRIPTION
**Pull request by marioevz**
## Changes Included
- Updates json field names to match the names used in https://github.com/lightclient/go-ethereum/tree/devnet-8
- Verify header fields of the execution result returned from t8n (t8n branch had a bug where it was incorrectly adding only one blob per tx regardless of the actual number of blobs included)
- Add blob_gas_used and excess_blob_gas field verification to all 4844 tests.

### CODE REVIEW

File: src/ethereum_test_tools/common/types.py [hunk 1]
```diff
@@ -2222,6 +2226,30 @@ def join(self, modifier: Header) -> "FixtureHeader":
...
+                value = getattr(self, field_name)
+                if baseline_value is Header.EMPTY_FIELD:
+                    assert value is None, f"invalid header field {header_field}"
+                    continue
+                metadata = header_field.metadata
+                assert metadata is not None, f"Field {field_name} has no header field metadata"
+                field_metadata = metadata.get("source")
+                assert isinstance(field_metadata, HeaderFieldSource), (
+                    f"Field {field_name} has invalid header_field " f"metadata: {field_metadata}"
+                )
```
Comment (spencer-tb) [inline src/ethereum_test_tools/common/types.py:2247]: Maybe this is not needed as we do the same check within `collect()`.

**Review by spencer-tb — CHANGES_REQUESTED** [review]
Nice way of checking the result of the t8n is as expect.

File: src/ethereum_test_tools/common/types.py [hunk 2]
```diff
@@ -1008,7 +1008,7 @@ class Environment:
     excess_blob_gas: Optional[NumberConvertible] = field(
         default=None,
         json_encoder=JSONEncoder.Field(
-            name="currentExcessBlobGas",
+            name="excessBlobGas",
```
[the remark this answers is not in the document]
Comment (marioevz) [reply src/ethereum_test_tools/common/types.py:1011]: Let me put it in the PR so we can ask lightclient about it, I think it should not be a big deal.

[the remark this answers is not in the document]
Comment (marioevz) [reply src/ethereum_test_tools/common/types.py:1011]: @spencer-tb It's the second commit in this PR now: https://github.com/lightclient/go-ethereum/pull/37

I'll remove the changes from this PR :+1:

File: src/ethereum_test_tools/common/types.py (hunk 1 above)
Comment (marioevz) [reply src/ethereum_test_tools/common/types.py:2247]: Removed :+1:

**Review by spencer-tb — APPROVED** [review]
LGTM :D

**Pull request merged**